src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.InlineContext.cs

```diff
@@ -0,0 +1,582 @@
+
+            private InlineMethodContext(
+                ImmutableArray<SyntaxNode> statementsToInsertBeforeCallee,
+                SyntaxNode statementContainingCallee,
+                SyntaxNode inlineSyntaxNode,
+                SyntaxNode syntaxNodeToReplace,
+                bool containsAwaitExpression)
+            {
+                StatementsToInsertBeforeCallee = statementsToInsertBeforeCallee;
+                StatementContainingCallee = statementContainingCallee;
+                InlineSyntaxNode = inlineSyntaxNode;
+                SyntaxNodeToReplace = syntaxNodeToReplace;
+                ContainsAwaitExpression = containsAwaitExpression;
+            }
+
+            public static async Task<InlineMethodContext> GetInlineContextAsync(
+                AbstractInlineMethodRefactoringProvider<TInvocationNode, TExpression, TArgumentSyntax, TMethodDeclarationSyntax, TIdentifierName> inlineMethodRefactoringProvider,
+                Document document,
+                SyntaxNode calleeInvocationNode,
+                IMethodSymbol calleeMethodSymbol,
+                TMethodDeclarationSyntax calleeMethodDeclarationNode,
+                SyntaxNode statementContainsCallee,
+                MethodParametersInfo methodParametersInfo,
+                CancellationToken cancellationToken)
+            {
+                var syntaxFacts = inlineMethodRefactoringProvider._syntaxFacts;
+                var semanticFactsService = inlineMethodRefactoringProvider._semanticFactsService;
+                var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                var syntaxGenerator = SyntaxGenerator.GetGenerator(document);
+                var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var inlineNode = inlineMethodRefactoringProvider.GetInlineStatement(calleeMethodDeclarationNode);
+
+                // Get the parameterSymbol to identifier's name mappings.
+                // Example:
+                // void Caller()
+                // {
+                //     int i = 10;
+                //     Callee(i);
+                // }
+                // void Callee(int a)
+                // {
+                //     DoSomething(a);
+                // }
+                // Input is an array of (ParameterSymbol(int a), IdentifierSyntaxNode(int i))
+                // Output is an array of (ParameterSymbol(int a), string i)
+                var parametersWithIdentifierArgumentToName = methodParametersInfo.ParametersWithIdentifierArgument
+                    .Select(parameterAndArgument =>
+                        (parameterAndArgument.parameterSymbol, semanticModel.GetSymbolInfo(parameterAndArgument.identifierSyntaxNode, cancellationToken).GetAnySymbol()?.Name))
+                    .Where(parameterAndName => parameterAndName.Name != null)
+                    .ToImmutableArray();
```

Comment:
WhereAsArray?

Reply:
Looks like WhereAsArray is only provided for ImmutableArray<T>? But here it is IEnumerable<T>. Did I misse any extension?